internal/merkle/inmemory/tree_test.go

```diff
@@ -16,634 +16,190 @@ package inmemory
 
 import (
 	"bytes"
-	"encoding/hex"
-	"errors"
 	"fmt"
 	"math/rand"
+	"strconv"
 	"testing"
 
 	_ "github.com/golang/glog"
-	"github.com/transparency-dev/merkle"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
 	"github.com/transparency-dev/merkle/rfc6962"
+	to "github.com/transparency-dev/merkle/testonly"
 )
 
-// TODO(pavelkalinnikov): Rewrite this file entirely.
-
-var fuzzTestSize = int64(256)
-
-// This is the hash of an empty string
-var emptyTreeHashValue = "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
-
-// Inputs to the reference tree, which has eight leaves.
-var leafInputs = [][]byte{
-	hx(""), hx("00"), hx("10"), hx("2021"), hx("3031"), hx("40414243"),
-	hx("5051525354555657"), hx("606162636465666768696a6b6c6d6e6f"),
-}
-
-// Incremental roots from building the reference tree from inputs leaf-by-leaf.
-// Generated from ReferenceMerkleTreeHash in C++.
-var rootsAtSize = []string{
-	"6e340b9cffb37a989ca544e6bb780a2c78901d3fb33738768511a30617afa01d",
-	"fac54203e7cc696cf0dfcb42c92a1d9dbaf70ad9e621f4bd8d98662f00e3c125",
-	"aeb6bcfe274b70a14fb067a5e5578264db0fa9b51af5e0ba159158f329e06e77",
-	"d37ee418976dd95753c1c73862b9398fa2a2cf9b4ff0fdfe8b30cd95209614b7",
-	"4e3bbb1f7b478dcfe71fb631631519a3bca12c9aefca1612bfce4c13a86264d4",
-	"76e67dadbcdf1e10e1b74ddc608abd2f98dfb16fbce75277b5232a127f2087ef",
-	"ddb89be403809e325750d3d263cd78929c2942b7942a34b77e122c9594a74c8c",
-	"5dc9da79a70659a9ad559cb701ded9a2ab9d823aad2f4960cfe370eff4604328",
-}
-
-// Some paths for the reference tree.
-type pathTestVector struct {
-	leaf       uint64
-	snapshot   uint64
-	testVector []string
-}
-
-// Generated from C++ ReferenceMerklePath, not the Go one so we can verify
-// that they are both producing the same paths in a sanity test.
```

Comment:
Can we move this comment and the other similar ones to the `testonly` package? It would helpful there for understanding where these constants were coming from.

Reply:
My plan is:
- Complete this PR.
- Move this code to `testonly` in https://transparency-dev/merkle.
- Use these and other constants as "golden" tests there.

Looks inline with your suggestion?

Reply:
SGTM